packages/next/build/analysis/get-page-static-info.ts

```diff
@@ -239,6 +240,23 @@ function warnAboutUnsupportedValue(
   warnedUnsupportedValueMap.set(pageFilePath, true)
 }
 
+function getUserConfig(
+  config: any,
+  middlewareConfig: Partial<MiddlewareConfig>
+) {
+  if (!config) return
+
+  const userConfig = config && { ...config }
+  if (userConfig) {
+    delete userConfig.runtime
+    if (middlewareConfig.matchers) {
+      delete userConfig.matcher
+    }
+  }
```

Comment:
nit: we already have an early return checking if `config` is falsy

```suggestion
  const userConfig = { ...config }
  delete userConfig.runtime
  if (middlewareConfig.matchers) {
    delete userConfig.matcher
  }
```